src/service/viewport-impl.js

```diff
@@ -414,7 +410,17 @@ export class Viewport {
       });
     }
 
-    return this.binding_.getLayoutRectAsync(el, scrollLeft, scrollTop);
+    // Use the most updated scrollTop and scrollLeft.
+    return this.binding_.getLayoutRectAsync(el);
+  }
+
+  getRectAsync() {
+    return this.vsync_.measurePromise(() => {
```

Comment:
This should return a memoized promise until the vsync executes (it'll save a lot of memory).